include/oneapi/tbb/detail/_segment_table.h

```diff
@@ -260,15 +260,19 @@ class segment_table {
         }
     }
 
+    void destroy_and_deallocate_table(segment_table_type table, size_type num_segments) {
```

Comment:
Is there a specific reason for having `num_segments` as a function parameter?
As far as I can see, `pointers_per_long_table` is used everywhere and it is a `static constexpr` value known by the class.

Reply:
This improves encapsulation of separate code pieces, allowing the function to deallocate segment tables of arbitrary size.

Reply:
OK, I don't mind.